### ISSUE TITLE
Allow more than 32 dynamic lights

### ISSUE THREAD

**Issue by Vac95** [Facepunch/garrysmod-requests#1273]
Currently you are only allowed to have 32 Dynamic Lights on a map

https://wiki.garrysmod.com/page/Global/DynamicLight

I was wonder if you would be able to allow people to manually raise this limit themselves.

**Comment by thegrb93** [comment]
Pretty sure its hardcoded into the shader

**Comment by Metal-Fish** [comment]
I might be total wrong but I believe gmod uses SDK 2013, which according to the valve [wiki ](https://developer.valvesoftware.com/wiki/Light_dynamic), says only 17 light_dynamic can be used. However it appears gmod somehow upped this number to 32.

**Comment by bigdogmat** [comment]
Going off [this](https://github.com/ValveSoftware/source-sdk-2013/blob/0d8dceea4310fde5706b3ce1c70609d72a38efdf/sp/src/public/iefx.h#L22) the sdk limit is 32.

**Comment by ExtReMLapin** [comment]
It only uses some parts of SDK 2013 @Metal-Fish

**Comment by Kefta** [comment]
I think the dlight and elight limits should both be raised to 128 as to match the player count. That way, servers could in essence give every player one light for gun muzzle flashes and gmod_lights.

**Comment by WinterPhoenix96** [comment]
I agree that the limits should be raised. I currently have an issue where 32 dlights is an incredibly small limit for my use case, and at 32 isn't impacting performance particularly severely.

128 would be a great start, at the very least.

**Comment by NathanX-S** [comment]
Bumping this. @robotboy655

**Comment by Roachdaripper** [comment]
bumping this @robotboy655
